x-pack/solutions/observability/plugins/apm/server/lib/apm_telemetry/collect_data_telemetry/tasks.ts

```diff
@@ -1180,132 +1175,115 @@ export const tasks: TelemetryTask[] = [
               },
             },
           });
-
-          const { aggregations } = response;
-
-          if (!aggregations) {
+  
+          if (!response.aggregations) {
             return data;
           }
-
-          const initAgentData = OPEN_TELEMETRY_AGENT_NAMES.reduce((acc, agent) => {
-            acc[agent] = {
+  
+          const dynamicAgentData: NonNullable<APMTelemetry['agents']> = {};
+  
+          for (const agentBucket of response.aggregations.agent_name.buckets) {
+            const agentKey = agentBucket.key as string;
+  
+            dynamicAgentData[agentKey] = {
               agent: {
-                activation_method: [],
-                version: [],
-              },
-              service: {
-                framework: {
-                  name: [],
-                  version: [],
-                  composite: [],
-                },
-                language: {
-                  name: [],
-                  version: [],
-                  composite: [],
-                },
-                runtime: {
-                  name: [],
-                  version: [],
-                  composite: [],
-                },
-              },
-            };
-            return acc;
-          }, {} as NonNullable<APMTelemetry['agents']>);
-
-          const agentData = aggregations?.agent_name.buckets.reduce((acc, agentNamesAggs) => {
-            acc[agentNamesAggs.key as OpenTelemetryAgentName] = {
-              agent: {
-                activation_method: agentNamesAggs[AGENT_ACTIVATION_METHOD].buckets
-                  .map((bucket) => bucket.key as string)
+                activation_method: agentBucket[AGENT_ACTIVATION_METHOD].buckets
+                  .map((bucket: any) => bucket.key as string)
```

Comment:
Could you please avoid using the type any. In this case you set it any and then cast to string.